pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -103,6 +128,48 @@ func GetDockercfgFile(path string) string {
 	return cfgPath
 }
 
+func readSpecificDockerConfigJSONFile(filePath string) bool {
+	var contents []byte
+	var err error
+
+	if contents, err = ioutil.ReadFile(filePath); err != nil {
+		log.V(4).Infof("error reading file: %v", err)
+		return false
+	}
+	return readDockerConfigJSONFileFromBytes(contents)
+}
+
+func readDockerConfigJSONFileFromBytes(contents []byte) bool {
+	var cfgJSON credentialprovider.DockerConfigJson
+	if err := json.Unmarshal(contents, &cfgJSON); err != nil {
+		log.V(4).Infof("while trying to parse blob %q: %v", contents, err)
+		return false
+	}
+	return true
+}
+
+// GetDockerConfigPath returns the first path that provides a valid DockerConfig;
+// modified elements from credentialprovider methods called from GetDockerConfig, following the same order of precedenced
+// articulated in GetDockerConfig, via the order of file names set in dockerFilesToExamine
+func GetDockerConfigPath(paths []string) string {
+	for _, configPath := range paths {
+		for _, file := range dockerFilesToExamine {
+			absDockerConfigFileLocation, err := filepath.Abs(filepath.Join(configPath, file))
+			if err != nil {
+				log.V(4).Infof("while trying to canonicalize %s: %v", configPath, err)
+				continue
+			}
+			klog.V(4).Infof("looking for %s at %s", file, absDockerConfigFileLocation)
```

Comment:
nit - why is this using `klog` instead of `log`?

Reply:
will fix momentarily

Reply:
update pushed @adambkaplan thanks